keyserver/src/main/scala/com/advancedtelematic/tuf/keyserver/daemon/KeyGeneratorLeader.scala

```diff
@@ -102,17 +103,16 @@ object DefaultKeyGenerationOp {
 class DefaultKeyGenerationOp(vaultClient: VaultClient)(implicit val db: Database, val ec: ExecutionContext)
   extends KeyGenerationOp
     with KeyGenRequestSupport
-    with KeyRepositorySupport
-    with RoleRepositorySupport {
+    with KeyRepositorySupport {
 
   private lazy val _log = LoggerFactory.getLogger(this.getClass)
 
-  protected def generateKeys(roleId: RoleId, keyType: KeyType, keySize: Int, threshold: Int): Seq[(Key, TufKeyPair)] = {
-    require(threshold > 0, "threshold must be greater than 0")
+  protected def generateKeys(kgr: KeyGenRequest): Seq[(Key, TufKeyPair)] = {
```

Comment:
One thing about this is that we specifically don't want redundant keys are the same hardware/device. The tool should aim to support metadata generated on multiple laptops and less on making many keys on one laptop. This isn't exactly a blocker, but it makes me a tad nervous about misuse.

Reply:
What do you mean? I don't understand how your comment is related with this change?

Reply:
How are we emitting a sequence of keys? Wouldn't that imply multiple private keys exist on a single machine?

Reply:
Here we generate one key for each role, so a Seq[Key]